spring-cloud-gcp-logging/src/main/java/org/springframework/cloud/gcp/logging/PrioritizedMultipleHeaderTraceIdFromRequestExtractor.java

```diff
@@ -0,0 +1,62 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.logging;
+
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.springframework.util.Assert;
+
+/**
+ * Checks HTTP request for multiple headers that might contain a trace ID, and provides
+ * the first one.
+ *
+ * @author Chengyuan Zhao
+ */
+public class PrioritizedMultipleHeaderTraceIdFromRequestExtractor
+		implements TraceIdFromRequestExtractor {
+
+	private List<TraceIdFromRequestExtractor> subExtractors;
```

Comment:
Well, usually we call them `target` or `delegate`, but they definitely are not `sub`. Because I don't see that they are doing some substruction from the result.